cirq/experiments/xeb_theory.ipynb

```diff
@@ -0,0 +1,404 @@
+    "Remember:\n",
+    "\n",
+    "$$\n",
+    "    m_U - u_U = f (e_U - u_U)\n",
+    "$$\n",
+    "\n",
+    "We estimate f by performing least squares\n",
+    "minimization of the quantity\n",
+    "\n",
+    "$$\n",
+    "    f (e_U - u_U) - (m_U - u_U)\n",
+    "$$\n",
+    "\n",
+    "over different random circuits. The solution to the\n",
+    "least squares problem is given by\n",
+    "\n",
+    "$$\n",
+    "    f = (∑_U (m_U - u_U) * (e_U - u_U)) / (∑_U (e_U - u_U)^2)\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "df = pd.DataFrame(records)\n",
+    "df['y'] = df['m_u'] - df['u_u']\n",
+    "df['x'] = df['e_u'] - df['u_u']\n",
+    "\n",
+    "df['numerator'] = df['x'] * df['y']\n",
+    "df['denominator'] = df['x'] ** 2\n",
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Fit\n",
+    "\n",
+    "We'll plot the linear relationship and least-squares fit while we transform the raw DataFrame into one containing fidelities."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "from matplotlib import pyplot as plt\n",
+    "\n",
+    "# Color by cycle depth\n",
+    "import seaborn as sns\n",
+    "colors = sns.cubehelix_palette(n_colors=len(cycle_depths))        \n",
+    "colors = {k: colors[i] for i, k in enumerate(cycle_depths)}\n",
+    "\n",
+    "_lines = []\n",
+    "def per_cycle_depth(df):\n",
+    "    fid_lsq = df['numerator'].sum() / df['denominator'].sum()\n",
+    "    \n",
+    "    cycle_depth = df.name    \n",
+    "    xx = np.linspace(0, df['x'].max())\n",
+    "    l, = plt.plot(xx, fid_lsq*xx, color=colors[cycle_depth])\n",
+    "    plt.scatter(df['x'], df['y'], color=colors[cycle_depth])\n",
+    "    \n",
+    "    global _lines\n",
+    "    _lines += [l] # for legend\n",
+    "    return pd.Series({'fid_lsq': fid_lsq})\n",
+    "\n",
+    "fids = df.groupby('cycle_depth').apply(per_cycle_depth).reset_index()\n",
+    "plt.xlabel(r'$e_U - u_U$', fontsize=18)\n",
+    "plt.ylabel(r'$m_U - u_U$', fontsize=18)\n",
+    "_lines = np.asarray(_lines)\n",
+    "plt.legend(_lines[[0,-1]], cycle_depths[[0,-1]], loc='best', title='Cycle depth')\n",
+    "plt.tight_layout()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Fidelities"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.plot(fids['cycle_depth'], fids['fid_lsq'], label='LSq')\n",
+    "\n",
+    "xx = np.linspace(0, fids['cycle_depth'].max())\n",
+    "plt.plot(xx, (1-P_DEPOL)**(4*xx), label=r'$(1-\\mathrm{depol})^{4d}$')\n",
+    "\n",
+    "plt.ylabel('Circuit fidelity', fontsize=18)\n",
+    "plt.xlabel('Cycle Depth', fontsize=18)\n",
```

Comment:
'Cycle dept d' to connect with {4d} in the label above?

Reply:
done